engine/src/main/java/org/camunda/bpm/engine/repository/ProcessDefinitionQuery.java

```diff
@@ -80,7 +80,7 @@ public interface ProcessDefinitionQuery extends Query<ProcessDefinitionQuery, Pr
   /**
    * Only select process definitions with the given keys
    */
-  ProcessDefinitionQueryImpl processDefinitionKeysIn(String... processDefinitionKeys);
+  ProcessDefinitionQuery processDefinitionKeysIn(String... processDefinitionKeys);
```

Comment:
Unfortunately, this is part of the public API, a change of the return type might break existing code for users and we have public API guaranties.
I see that there's a point, the method should have been written that way in the beginning, however to be more user friendly, my suggestion is to create a new method and deprecate the old one.
As this method is the only one in the queries to have plural keys, I also see room for improving the consistency and name the new method `#processDefinitionKeyIn`. What do you think?
```suggestion
  /**
   * Only select process definitions with the given keys
   * <p><b>Deprecated</b>: use {@link #processDefinitionKeyIn(String...)}</p>
   */
  ProcessDefinitionQueryImpl processDefinitionKeysIn(String... processDefinitionKeys);

  /**
   * Only select process definitions with the given keys
   */
  ProcessDefinitionQuery processDefinitionKeyIn(String... processDefinitionKeys);
```

Reply:
I fixed that. You are 100% right...